### ISSUE TITLE
Missing space before new line cause SQL error with MariaDB

### ISSUE THREAD

**Issue by afterlastangel** [advanced-online-marketing/AOM#34]
AOM version 1.0.6
Database: 10.1.8-MariaDB
PHP Version: 7.0.14
```
ERROR: Failed to get data from API: Mysqli prepare error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '?GROUP BY campaignId' at line 1 #0 =

Mysqli prepare error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '?GROUP BY campaignId' at line 1
```
The log trace lead to 
plugins/AOM/Platforms/AdWords/MarketingPerformanceSubTables.php(85): Piwik\Db::fetchAssoc('SELECT CONCAT('...', Array)
<img width="943" alt="screen shot 2017-07-12 at 10 25 09 pm" src="https://user-images.githubusercontent.com/1694449/28125519-03ad0c1a-6751-11e7-87e1-95e62662b83c.png">

Resolution:

Add more trailing space into the SQL query on MarketingPerformanceSubTables.php

I will make a pull request into the develop branch.

**Comment by afterlastangel** [comment]
<img width="967" alt="screen shot 2017-07-13 at 6 06 05 am" src="https://user-images.githubusercontent.com/1694449/28144956-e04bcd22-6798-11e7-8e10-8e6ceeadef3e.png">
I still with the subtable of subtable, but it does not show any data after fix it. I will debug more

**Comment by andre-hh** [comment]
I just release 1.0.8 which fixes some issues within the subtables. Can you check again?

One issue that is still open, is the formatting of some numbers. Currently no revenue is shown in that table, as there were warnings being generated. If you finde some time to investigate, that would be awesome. Otherwise I'll do it within the next days hopefully.

**Comment by afterlastangel** [comment]
@andre-hh I have a freelance task on fixing these problems, the client also happy to make the PR to contribute back to opensource version. So let me investigate this problem. 
Please mind that the upgrade process from version 0.6 to 1.0 has some problem with the database schema (missing unique_hash, platform_key in aom_visits), as Piwik does not support migration for plugins. So that we should do database schema migration manually if we want to keep the old data.

**Comment by andre-hh** [comment]
You are absolutely right, there were some database changes required. When upgrading from 0.6 to 1.0, it is no longer possible to use AdWords without gclid. There are also some new database columns, leading to the problem that these columns won't have any data in them for import and merge jobs than ran before the upgrade, i.e. you should not rerun imports & merges that ran before the upgrade (this data might not be correct then).

I hope that we'll be able to not have any breaking changes with 1.0 now.

We might move the aom:process command that should run as a cronjob currently every minute to a long running background job (e.g. managed with supervisor). Also the upcoming feature of individual campaigns will add a new table / modify that new table.

I've updated the docs at advanced-online-marketing.com recently.

**Comment by afterlastangel** [comment]
hello @andre_hh, I found the problem with the subtable. With platform_data format, we should change the sql query

**Comment by andre-hh** [comment]
Thank you very much for your PR. I just reviewed it. I think the right way is to have strings instead of numbers in the JSON to avoid integer overflows. We might temporarily support all three types of JSON... 

The process cron should fill the piwik_idvisitor field with the idvisitor from the log_visits table. Perhaps it seems as if there is no value in the field as it is binary data? The usual process should be, that the piwik_aom_visits table is filled every minute with new visits from the log_visits table and that the other background jobs allocate cost and add campaign data to these visits.

**Comment by andre-hh** [comment]
With the latest version of Piwik AOM various (big) integers should be stored as strings.
Can you confirm this?

**Comment by afterlastangel** [comment]
Hi @andre-hh, I would like to confirm that the Problem of Big Integers Fixed, it's stored as string now. 
I added a cron to run aom:process for every minutes. There is still some SQL issue with conversion
mapping. I would investigate later and make another PR for this problem.
```
  Mysqli prepare error: Unknown column 'idconversion' in 'field list'
```

**Comment by andre-hh** [comment]
@afterlastangel, thanks for your confirmation.

When you take a look at https://github.com/advanced-online-marketing/AOM/blob/master/AOM.php#L93 a fresh install of the AOM plugin would add this column to Piwik.

**Comment by afterlastangel** [comment]
@andre-hh Everything works well right now. I would close this issues after this PR is approved 
https://github.com/advanced-online-marketing/AOM/pull/38
There are some places missing space before `group by`.
If you have any task or issue according AOM, you might assign It to me, I'm willing to contribute to AOM project.

**Comment by andre-hh** [comment]
I have just merged the PR and released version 1.1.3. 
If you like to and have access to Facebook Ads, you could reimplement that?
Any other things you miss in Piwik AOM or would like to see? Then we could align about them and implement them.
